Mobile Navbar: Fix dropdown links + Close on click off

a-ys opened the pull request:
## First 2 Commits
Issue: for the mobile navbar, when it was closed, the "For Students" and "For Nonprofits" were invisible but clickable.
Other changes:
    - rename variables associated with navbar state
    - fix active classname of navMenu. See [this comment for details.](https://github.com/Hack4Impact-UMD/umd-website-redesign/commit/536aa742dc09b4867b6956877be36969099259ef#r80706729)

## Last Commit
Issue: want to add click on whitespace to close mobile navbar.
My implementation is not complete. There are three areas in the expanded mobile navbar: 
1. whitespace around links (padding, margins, etc). 
2. the text itself. 
3. whitespace below text. 

I added the click off for only area 3. @heylookitsme I think you will be adding this for area 2 (clicking on a link and navigating to another page). For area 1, it's a bit more tricky I think because there's a lot of padding/margins for the list elements.

a-ys committed:
Add background blur for navbar

yieb committed:
Change style id names for clarity

a-ys committed:
Fix mobile navbar apply sublinks
The links were previously clickable while invisible when the navbar was closed.
Other changes:
    - rename variables associated with navbar state
    - fix active classname of navMenu.

a-ys committed:
Fix mobile navbar apply sublinks
The links were previously clickable while invisible when the navbar was closed.
Other changes:
    - rename variables associated with navbar state
    - fix active classname of navMenu.

a-ys committed:
reformat navbar component and styles

a-ys committed:
reformat navbar component and styles

a-ys committed:
Mobile Navbar: close on click off (partial)

a-ys committed:
Mobile Navbar: close on click off (partial)

heylookitsme commented:
ahm 
![sob-emoji](https://user-images.githubusercontent.com/45301066/184119760-a4703603-6f3b-4a3b-9131-82f538d6bc32.gif)

Commit:
Merge branch 'mobile_navbar_dropdown_links_fix' of https://github.com/Hack4Impact-UMD/umd-website-redesign into mobile_navbar_dropdown_links_fix

Commit:
Improved Navbar toggle, addressed animation bug, styled

possibly the worst commit ive ever made. will elaborate in PR
retrospectivley.

TODO: MUST REDO THIS

Commit:
Merge branch 'main' into mobile_navbar_dropdown_links_fix
